sklearn/linear_model/ridge.py

```diff
@@ -1015,48 +1164,168 @@ def _errors_and_values_helper(self, alpha, y, v, Q, QT_y):
             G_diag = G_diag[:, np.newaxis]
         return G_diag, c
 
-    def _errors(self, alpha, y, v, Q, QT_y):
-        G_diag, c = self._errors_and_values_helper(alpha, y, v, Q, QT_y)
+    def _errors_gram(self, alpha, y, v, Q, QT_y):
+        G_diag, c = self._errors_and_values_gram(alpha, y, v, Q, QT_y)
         return (c / G_diag) ** 2, c
 
-    def _values(self, alpha, y, v, Q, QT_y):
-        G_diag, c = self._errors_and_values_helper(alpha, y, v, Q, QT_y)
+    def _values_gram(self, alpha, y, v, Q, QT_y):
+        G_diag, c = self._errors_and_values_gram(alpha, y, v, Q, QT_y)
         return y - (c / G_diag), c
 
-    def _pre_compute_svd(self, X, y, centered_kernel=True):
-        if sparse.issparse(X):
-            raise TypeError("SVD not supported for sparse matrices")
-        if centered_kernel:
-            X = np.hstack((X, np.ones((X.shape[0], 1))))
+    def _decompose_covariance_sparse(self, X, y):
```

Comment:
You can put a small docstring (single line) stating for which case we should use it `n_samples` > `n_features`